### DISCUSSION TITLE
What's a good way to list transactions?

### DISCUSSION THREAD

**Discussion by gd87429** [rust-bitcoin/rust-bitcoin#1382] [Q&A] [answered]
Given a block hash or height, what's a good way to list transactions within? Didn't see sample code to get started with in [examples](https://github.com/rust-bitcoin/rust-bitcoin/tree/master/bitcoin/examples). Could someone point me in the right direction?

**Comment by apoelstra** [comment, marked as answer]
The easiest way, I think, would be to get the block, in hex, from the Bitcoin Core RPC by using `bitcoin-cli getblock $(bitcoin-cli getblockhash $HASH) 0`.

Then using this library, you can parse the hex using [`deserialize`](https://docs.rs/bitcoin/latest/bitcoin/consensus/encode/fn.deserialize.html) to get a `bitcoin::Block`, then iterate over the `txdata` field of that. You will first need to deserialize the hex into a `Vec<u8>` using the `bitcoin::hashes::hex::FromHex` trait.

We really should expose a `deserialize_hex` method at the top level of this library..

**Reply by gd87429** [reply]
Perfect. Thank you, I'll try that.
